### ISSUE TITLE
<depends> field doesn't work in system.xml for "radios" fields

### ISSUE THREAD

**Issue by WaPoNe** [magento/magento2#9360]
In system.xml file a field can depend on another field value; it will be shown or not depending from previous field choice.
It works fine if the field depends on a "select" field but if you try to use a "radios" field dependency the mechanism doesn't work and the field will always be shown.

### Preconditions

1. Magento 2.1.5

### Steps to reproduce

**app/code/vendorName/moduleName/etc/adminhtml/system.xml**:

```
<?xml version="1.0"?>
<config xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:noNamespaceSchemaLocation="urn:magento:module:Magento_Config:etc/system_file.xsd">
    <system>
        <tab id="wapone" translate="label" sortOrder="1000">
            <label>WaPoNe</label>
        </tab>
        <section id="stickers" translate="label" type="text"
                 sortOrder="100" showInDefault="1" showInWebsite="1" showInStore="1">
            <label>Stickers</label>
            <tab>wapone</tab>
            <resource>WaPoNe_Stickers::config_stickers</resource>
            <group id="stickers_page" translate="label" type="text"
                   sortOrder="1" showInDefault="1" showInWebsite="1" showInStore="1">
                <label>Stickers page settings</label>
                <field id="discount_calculation" translate="label" type="radios"
                       sortOrder="1" showInDefault="1" showInWebsite="1" showInStore="1">
                    <label>Discount Calculation</label>
                    <source_model>WaPoNe\Stickers\Model\Config\Backend\DiscountCalculationType</source_model>
                </field>
                <field id="discount_category" translate="label" type="text"
                       sortOrder="2" showInDefault="1" showInWebsite="1" showInStore="1">
                    <label>Discount Product Category</label>
                    <depends>
                        <field id="stickers/stickers_page/discount_calculation">automatic</field>
                    </depends>
                </field>
            </group>
        </section>
    </system>
</config>
```

**app/code/vendorName/moduleName/Model/Config/Backend/DiscountCalculationType.php**:

```
<?php

namespace WaPoNe\Stickers\Model\Config\Backend;

use Magento\Framework\Option\ArrayInterface;

class DiscountCalculationType implements ArrayInterface
{
    public function toOptionArray()
    {
        $calculationType = array();

        $calculationType[] = [
            'value' => 'manual',
            'label' => __('Manual')
        ];
        $calculationType[] = [
            'value' => 'automatic',
            'label' => __('Automatic')
        ];

        return $calculationType;
    }
}
```

### Expected result

1. _discount_calculation_ value = 'automatic' => _discount_category_ is shown
2. _discount_calculation_ value = 'manual' => _discount_category_ is not shown

### Actual result

1. _discount_category_ is always shown so <depend> is not working

If I declare _discount_calculation_ field as "select" type, all works fine

**Comment by magento-engcom-team** [comment]
@WaPoNe, thank you for your report.
We've created internal ticket(s) MAGETWO-80952 to track progress on the issue.

**Comment by osrecio** [comment]
Assign to me

**Comment by jahvi** [comment]
I started working on this issue over at #11539, feel free to comment on it.

**Comment by okorshenko** [comment]
The issue has been fixed and delivered to the 2.2-develop branch. Will be available with upcoming patch release

**Comment by magento-engcom-team** [comment]
Hi @WaPoNe. Thank you for your report. 
The issue has been fixed in magento-engcom-team/magento2ce#114 by @magento-engcom-team in 2.3-develop branch
Related commit(s):
 - [da5692cda81eff0d7dce889fac1c68ba4acd5ed7](https://github.com/magento/magento2/commit/da5692cda81eff0d7dce889fac1c68ba4acd5ed7)
 - [9775cd8e6fc94927d62e98d1e152735563f08c51](https://github.com/magento/magento2/commit/9775cd8e6fc94927d62e98d1e152735563f08c51)
 - [e10a3ff19ef1111171f4c2a89c2d3ab2f0f4d30f](https://github.com/magento/magento2/commit/e10a3ff19ef1111171f4c2a89c2d3ab2f0f4d30f)

The fix will be available with the upcoming 2.3.0 release.

**Comment by WaPoNe** [comment]
@magento-engcom-team great job!
